### PR TITLE
BIG-31415 Fix deprecation warnings and upgrade PHPUnit

### PR DESCRIPTION
**Pull request by TomA-R**
## What
- Fix deprecation warnings like `Creation of dynamic property Net_Http_Response::$headers is deprecated`
- Upgrade PHPUnit
- Add CircleCI config

ping @bigcommerce/php

### CODE REVIEW

File: test/ClientTest.php [hunk 1]
```diff
@@ -11,25 +11,27 @@
...
 		$client->post(self::HOST.'/basic/post', array("greeting"=>"Hello", "from"=>"Net_Http_Client"));
 		$this->assertEquals(200, $client->getStatus());
-		$this->assertContains("Hello back", $client->getBody());
+		$this->assertStringContainsString("Hello back", $client->getBody());
 	}
 
 	public function testHeadRequestWithSetHeader()
 	{
+		// The host used for these tests no longer returns the expected headers
+		$this->markTestSkipped();
```
Comment (PascalZajac) [inline test/ClientTest.php:33]: Given this library hasn't been touched in 9 years, I wonder if there's any value keeping these tests if they don't work...

Comment (TomA-R) [reply test/ClientTest.php]: I've deleted them since it's unlikely that we'll fix the host

File: src/Net/Http/Client.php [hunk 2]
```diff
@@ -107,6 +107,11 @@ class Net_Http_Client {
 	 */
 	private $responseInfo = array();
 
+	/**
+	 * @var false
```
Comment (funivan) [inline src/Net/Http/Client.php:111]: I think you can drop this comment.

**Review by funivan — APPROVED** [review]

File: src/Net/Http/Client.php (hunk 2 above)
Comment (TomA-R) [reply src/Net/Http/Client.php]: 👍 done

**Pull request merged**